docs/website/docs/general-usage/data-enrichments/user_agent_device_data_enrichment.md

```diff
@@ -0,0 +1,311 @@
+---
+title:  User-agent device data enrichment
+description: Enriching the user-agent device data with average device price.
+keywords: [data enrichment, user-agent data, device enrichment]
+---
+
+# Data enrichment part one: User-agent device data enrichment
+
+Data enrichment enhances raw data with valuable information from multiple sources, increasing its
+analytical and decision-making value.
+
+This part covers enriching the above sample data with device price. Understanding the price segment
+of the device that the user used to access your service can be helpful in “personalized marketing”,
+“customer segmentation,” and many more.
+
+This documentation will discuss how to enrich the user device information with the average market
+price.
+
+## Setup Guide
+
+We use SerpApi to retrieve device prices using Google Shopping, but alternative services or APIs are
+viable.
+
+:::note
+SerpApi free tier offers 100 free calls monthly. For production, consider upgrading to a higher
+plan.
+:::
+
+
+## Creating data enrichment pipeline
+You can either follow the example in linked Colab notebook or follow this documentation to
+create the user-agent device data enrichment pipeline.
+
+### A. Colab notebook
+The Colab notebook combines three data enrichment processes for a sample dataset, starting with "Data
+enrichment part one: User-agent device data".
+
+The first step is to register on [SerpApi](https://serpapi.com/) and obtain the
+API token key. To set up credentials in Colab secrets:
+1. In the Colab notebook, click on "Colab Secrets" in the left sidebar.
+2. Configure variable `serp_api_key` as with obtained API key.
```

Comment:
could you move all this section about Colab right into Colab? it should be self-contained and self-declarative notebook. You can only link colab for example here:

>You can either follow the example in **linked Colab** notebook or follow this documentation to

Reply:
moved